common/src/main/java/com/alibaba/nacos/common/executor/ThreadPoolManager.java

```diff
@@ -0,0 +1,182 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.alibaba.nacos.common.executor;
+
+
+import com.alibaba.nacos.common.utils.ShutdownUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * // TODO Access Metric
+ *
+ * For unified management of thread pool resources, the consumer can simply call
+ * the register method to {@link ThreadPoolManager#register(String, String, ExecutorService)}
+ * the thread pool that needs to be included in the life cycle management of the resource
+ *
+ * @author <a href="mailto:liaochuntao@live.com">liaochuntao</a>
+ */
+public final class ThreadPoolManager {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(ThreadPoolManager.class);
+
+    private Map<String, Map<String, Set<ExecutorService>>> resourcesManager;
+
+    private Map<String, Object> lockers = new ConcurrentHashMap<String, Object>(8);
+
+    private static final ThreadPoolManager INSTANCE = new ThreadPoolManager();
+
+    private static final AtomicBoolean CLOSED = new AtomicBoolean(false);
+
+    static {
+        INSTANCE.init();
+		ShutdownUtils.addShutdownHook(new Thread(new Runnable() {
+            @Override
+            public void run() {
+				LOGGER.warn("[ThreadPoolManager] Start destroying ThreadPool");
+                shutdown();
+				LOGGER.warn("[ThreadPoolManager] Destruction of the end");
+            }
+        }));
+    }
+
+    public static ThreadPoolManager getInstance() {
+        return INSTANCE;
+    }
+
+    private ThreadPoolManager() {}
+
+    private void init() {
+        resourcesManager = new ConcurrentHashMap<>(8);
+    }
+
+    /**
+	 * Register the thread pool resources with the resource manager
+	 *
+     * @param namespace namespace name
+	 * @param group group name
+	 * @param executor {@link ExecutorService}
+	 */
+	public void register(String namespace, String group, ExecutorService executor) {
+        synchronized(this) {
+            if (!resourcesManager.containsKey(namespace)) {
+                resourcesManager.put(namespace, new HashMap<String, Set<ExecutorService>>(8));
+                lockers.put(namespace, new Object());
+            }
+        }
+        final Object monitor = lockers.get(namespace);
+        synchronized (monitor) {
+            Map<String, Set<ExecutorService>> map = resourcesManager.get(namespace);
```

Comment:
it's no guarantee that the map would not be null，it's may be delete by other thread when concurrent

Reply:
yes, I already fix it